### ISSUE TITLE
DownloadPost autocommand

### ISSUE THREAD

**Issue by noctuid** [cmcaine/tridactyl#807]
Pentadactyl has an event for after something has been downloaded. This particularly useful to pass the file location to a shell script and then open a vim-like file manager (e.g. ranger) to choose a new location for the file.

**Comment by antonva** [comment]
I'm definitely not opposed to this idea, the API doesn't provide a clear way to do this though. We do have the [onChanged](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/downloads/onChanged) event though, maybe that fires when the download is finished?

**Comment by noctuid** [comment]
The example on the bottom of that pages seems to be how you would check if the download is completed. It looks like the data for the filename and url are available as well.

**Comment by antonva** [comment]
You are correct. I didn't look at the delta object offered. This is definitely doable.

**Comment by glacambre** [comment]
The betas have a `:saveas` command which accepts arbitrary paths (e.g. /tmp or ~/documents). It's not a proper downloadpost autocommand but it'll let you move stuff around.
